integration-tests/tests/src/tests/results.ts

```diff
@@ -0,0 +1,1060 @@
+        });
+      };
+
+      objects = objects.sorted("primaryCol");
+      expect(primaries(objects)).deep.equal([0, 1, 2, 3, 4]);
+
+      objects = objects.sorted("primaryCol", true);
+      expect(primaries(objects)).deep.equal([4, 3, 2, 1, 0]);
+
+      objects = objects.sorted(["primaryCol", "valueCol"]);
+      expect(primaries(objects)).deep.equal([0, 1, 2, 3, 4]);
+
+      objects = objects.sorted([
+        ["primaryCol", true],
+        ["valueCol", true],
+      ]);
+      expect(primaries(objects)).deep.equal([4, 3, 2, 1, 0]);
+
+      objects = objects.sorted([["primaryCol", false], "valueCol"]);
+      expect(primaries(objects)).deep.equal([0, 1, 2, 3, 4]);
+
+      objects = objects.sorted(["valueCol", "primaryCol"]);
+      expect(primaries(objects)).deep.equal([2, 3, 1, 0, 4]);
+
+      objects = objects.sorted([
+        ["valueCol", false],
+        ["primaryCol", true],
+      ]);
+      expect(primaries(objects)).deep.equal([3, 2, 1, 4, 0]);
+
+      objects = objects.sorted([
+        ["valueCol", true],
+        ["primaryCol", false],
+      ]);
+      expect(primaries(objects)).deep.equal([0, 4, 1, 2, 3]);
+
+      objects = objects.sorted([
+        ["valueCol", true],
+        ["primaryCol", true],
+      ]);
+      expect(primaries(objects)).deep.equal([4, 0, 1, 3, 2]);
+
+      expect(() => {
+        objects.sorted();
+      }).throws("property 'IntPrimaryObject.self' does not exist.");
+
+      expect(() => {
+        //@ts-expect-error Expected to be an invalid sorted argument.
+        objects.sorted(1);
+      }).throws("JS value must be of type 'string'");
+      expect(() => {
+        //@ts-expect-error Expected to be an invalid sorted argument.
+        objects.sorted([1]);
+      }).throws("JS value must be of type 'string'");
+
+      expect(() => {
+        objects.sorted("fish");
+      }).throws("property 'IntPrimaryObject.fish' does not exist");
+      expect(() => {
+        objects.sorted(["valueCol", "fish"]);
+      }).throws("property 'IntPrimaryObject.fish' does not exist");
+
+      expect(() => {
+        //@ts-expect-error Expected to be an invalid sorted argument.
+        objects.sorted(["valueCol", "primaryCol"], true);
+      }).throws("Second argument is not allowed if passed an array of sort descriptors");
+    });
+
+    it("implements sorted correctly with all types", () => {
+      const realm = new Realm({ schema: [BasicTypesSchema] });
+      let objects = realm.objects<BasicTypesObject>("BasicTypesObject");
+
+      const decimals: BSON.Decimal128[] = [];
+      const oids: BSON.ObjectId[] = [];
+      const uuids: BSON.UUID[] = [];
+      for (let i = 0; i < 3; i++) {
+        decimals.push(Decimal128.fromString(`${i}`));
+        oids.push(new ObjectId(`0000002a9a7969d24bea4cf${i}`));
+        uuids.push(new UUID(`f7adc509-49be-466f-b3d5-449696d9a60${i}`));
+      }
+
+      realm.write(function () {
+        realm.create("BasicTypesObject", [
```

Comment:
This will probably need to be refactored as well.  I think we are deprecating creating objects with an array.  It's also not very readable 😅